daemon/graphdriver/fsdiff.go

```diff
@@ -11,29 +11,21 @@ import (
 	"github.com/docker/docker/pkg/ioutils"
 )
 
-// naiveDiffDriver takes a ProtoDriver and adds the
-// capability of the Diffing methods which it may or may not
-// support on its own. See the comment on the exported
-// NaiveDiffDriver function below.
-// Notably, the AUFS driver doesn't need to be wrapped like this.
-type naiveDiffDriver struct {
-	ProtoDriver
-}
-
-// NaiveDiffDriver returns a fully functional driver that wraps the
-// given ProtoDriver and adds the capability of the following methods which
+// NaiveDiffDriver takes a ProtoDriver and adds the
+// capability of the following methods which
 // it may or may not support on its own:
 //     Diff(id, parent string) (archive.Archive, error)
 //     Changes(id, parent string) ([]archive.Change, error)
 //     ApplyDiff(id, parent string, diff archive.Reader) (size int64, err error)
 //     DiffSize(id, parent string) (size int64, err error)
-func NaiveDiffDriver(driver ProtoDriver) Driver {
-	return &naiveDiffDriver{ProtoDriver: driver}
+// Notably, the AUFS driver doesn't need to be wrapped like this.
+type NaiveDiffDriver struct {
+	ProtoDriver
 }
 
 // Diff produces an archive of the changes between the specified
 // layer and its parent layer which may be "".
-func (gdw *naiveDiffDriver) Diff(id, parent string) (arch archive.Archive, err error) {
+func (gdw NaiveDiffDriver) Diff(id, parent string) (arch archive.Archive, err error) {
```

Comment:
@vbatts `NaiveDiffDriver` is a struct wrapping an interface so I don't see how the pointer receiver would help. But I'm fine either way.